pkg/client/connector/k8s_config.go

```diff
@@ -1,17 +1,50 @@
 package connector
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
 	"sort"
 
 	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/rest"
 
 	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/telepresenceio/telepresence/v2/pkg/iputil"
 )
 
+// The dns config part of the kubeconfigExtension that is read from the kubeconfig
+type dns struct {
+	// LocalIP is the address of the local DNS server. This entry is only
+	// used on Linux system that are not configured to use systemd.resolved and
+	// can be overridden by using the option --dns on the command line and defaults
+	// to the first line of /etc/resolv.conf
+	LocalIP iputil.IPKey `json:"local-ip,omitempty"`
+
+	// RemoteIP is the address of the cluster's DNS service. It will default
+	// to the IP of the kube-dns.kube-system or the dns-default.openshift-dns service.
+	RemoteIP iputil.IPKey `json:"remote-ip,omitempty"`
+
+	// ExcludeSuffixes are suffixes for which the DNS resolver will always fail (or fallback
+	// in case of the overriding resolver).
+	ExcludeSuffixes []string `json:"exclude-suffixes,omitempty"`
```

Comment:
What does "fail" mean in this context? Returning NXDOMAIN?

Reply:
Yes, it will return NXDOMAIN.